sklearn/utils/_param_validation.py

```diff
@@ -16,6 +17,12 @@
 from .validation import _is_arraylike_not_scalar
 
 
+class InvalidParameterError(Exception):
```

Comment:
Changing to `InvalidParameterError` may break existing code that is catching `ValueError`. To be backward compatible, I think this needs to inherit from `ValueError`:

```suggestion
class InvalidParameterError(ValueError):
```

With this inheritance, a lot of the tests can remain the same and look for `ValueError`.